### ISSUE TITLE
The result of your model you provided is a little wrong with the images download from the website

### ISSUE THREAD

**Issue by lisentao** [monniert/unicorn#1]
I am very appreciated with your work.
I download the images from the website.I found the car, bird, and horse model can not work well with the images.I can get the shape and the texture in the result, but the texture, even the colour is wrong.
And the motobike model works well.

**Comment by monniert** [comment]
Hi @lisentao, thanks for raising the issue!

Can you be more specific by sending an example result? Have you tried to run the car demo, and did it work? Have you followed the installation tutorial and/or checked that package versions are identical to the ones in `environment.yml`? 

Thanks, Tom

**Comment by lisentao** [comment]
Thinks for your reply.I run the code on CUDA11.1, python3.8, RTX3060, pytorch1.8.
I can run the car demo well.We can see in the last image.
![image](https://user-images.githubusercontent.com/49113628/169770427-d004d325-59d2-4425-b9e9-81ff49849c1f.png)
But some images from the website does not work well,especially the colour.We can see in the last image.
![image](https://user-images.githubusercontent.com/49113628/169770632-08adc81c-32a8-4098-8dfc-1f5e2a6ba6ce.png)
The last image is worked from the horse model.I think there has problem with the shape.
![image](https://user-images.githubusercontent.com/49113628/169772205-8fffde74-3580-41f7-8dbe-93d57abbbd0d.png)
The last image is worked from the bird model. I think there has problem with the texture(especially the colour)
![image](https://user-images.githubusercontent.com/49113628/169771294-961edb94-fe25-46ff-b8d6-41ea63e441d5.png)
The last image is worked from the moto model.I think it works well.
![image](https://user-images.githubusercontent.com/49113628/169771974-c78a4f0c-cdb0-4bcd-ac6a-1a576d474aa7.png)

**Comment by monniert** [comment]
Thanks for the details. The working examples show that your package versions are not the main issue, although it may be the cause of some differences.

In general, I think the non-working examples are either (i) out-of-distribution wrt to the training examples (e.g., there are no/very few cars like the sport one or the vintage one in CompCars, and the last bird has exotic colors) or (ii) very difficult in the case of the horse (but the horse model does not work well as outlined in the paper). We are currently working on small fixes to improve these cases in particular wrt the texture quality, I will let you know if we achieve better results in these scenarios.

**Comment by lisentao** [comment]
I am really appreciated for your quick reply.

**Comment by monniert** [comment]
Hi @lisentao, I just pushed a new version of the code with bigger models (corresponding to bigger decoders and bigger latent codes). In general, we found it improves a bit the texture quality but it is still giving poor results on these examples with complex textures. My hypothesis is that, although the swap/neighbor reconstruction loss is critical to learn realistic unseen parts, it somehow prevents the prediction of textures that are unique in the dataset. Finding a way to improve the texture quality (with loss weights finetuning, local features like PIFu or more advanced autoencoders like styleGAN) may be a good direction for future works.

<img width="256" alt="test1" src="https://user-images.githubusercontent.com/25098794/185349666-be3573a3-b8a7-41f7-a052-eaacd0b0f549.png">

![test1_mesh](https://user-images.githubusercontent.com/25098794/185349731-15b34866-3ce2-4fc4-b241-d4b57066ba88.gif)

<img width="256" alt="test2" src="https://user-images.githubusercontent.com/25098794/185350147-48a272a9-e08a-46ec-b1cd-4b90489c8a95.png">

![test2_mesh](https://user-images.githubusercontent.com/25098794/185350165-ece3e0af-a57c-46b3-855c-9e5a9ce84d65.gif)

<img width="256" alt="bird2" src="https://user-images.githubusercontent.com/25098794/185350200-07c50840-0c48-4789-adf6-81730b836bf4.png">

![bird2_mesh](https://user-images.githubusercontent.com/25098794/185350226-63871c5f-d645-4d3d-a2e4-c54d50ff4b57.gif)

**Comment by lisentao** [comment]
I got it.Thank you for your reply.
When I test the code of unicorn before,I just try to reconstruct  the boat with high quality and little time using just 4 permanent position camera.With lots of limits, I can not use the traditional codes like colmap.I have tried the single view reconstruction algorithm like UMR or SMR,but still can not get good results for boat in the wild.Now I use the code of NeRS(2021).I use four pictures and the result is barely satisfactory.
I think the code of single view/little view 3d reconstruction still have limits for complex and big object in the wild. And I can understand the difficuly.
I am really appreciate for your reply,and I think it provides a good reference for my master's thesis.
